rocketpy/Function.py

```diff
@@ -473,6 +473,92 @@ def setDiscrete(
             self.__interpolation__ = "shepard"
         return self
 
+    def setDiscreteBasedOnModel(self, modelFunction, oneByOne=True):
+        """This method transforms function defined Functions into list
+        defined Functions. It evaluates the function at certain points
+        (sampling range) and stores the results in a list, which is converted
+        into a Function and then returned. The original Function object is
+        replaced by the new one.
```

Comment:
I'm a little lost on this method. What is it supposed to do? And where can this be used?

Also, I think the docs here need to be changed, it is exactly the same as the `setDiscrete` method

Reply:
I will update the docstring and add usage examples. Thanks for letting me know it is not at all clear!

Reply:
I have updated the docs. Mark as resolved if you are satisfied.